src/types/ErrorRate.ts

```diff
@@ -0,0 +1,320 @@
+import { NA, RequestHealth, REQUESTS_THRESHOLDS } from './Health';
+import { RateHealthConfig, RegexConfig, ToleranceConfig } from './ServerConfig';
+import { serverConfig } from '../config/ServerConfig';
+import { ascendingThresholdCheck, ThresholdStatus, getRequestErrorsStatus, RATIO_NA, RequestType } from './Health';
+import { DecoratedGraphEdgeData, DecoratedGraphNodeData, ResponseDetail, Responses } from './Graph';
+
+export interface ErrorRatio {
+  global: { status: ThresholdStatus; protocol?: string };
+  inbound: { status: ThresholdStatus; protocol?: string };
+  outbound: { status: ThresholdStatus; protocol?: string };
+}
+
+export interface Rate {
+  requestRate: number;
+  errorRate: number;
+  errorRatio: number;
+}
+
+export const regexProtocol = {
+  grpc: new RegExp(/^[1-9]$|^1[0-6]$/),
+  /* eslint-disable-next-line */
+  http: new RegExp(/^[4-5]\d\d$/)
```

Comment:
We discussed making only 5xx the default for error thresholding. so that 404s would not cause a problem by default. If that is true then this regex needs to be updated.  I'm not sure we made a decision, just mentioning it here.

Reply:
The defaults are creating some problems, see my global comment, I'm +1 with this comment, but I think that it would be good to have the defaults explicit from the backend, otherwise it can be hard to debug, specially on these first iterations.

Reply:
I didn't remember that discussion. For me is fine to have only default for 500. The actual value is the health that we had before.

What do you think @lucasponce ?

Reply:
See my comment https://github.com/kiali/kiali-ui/pull/1831#issuecomment-676477436 where there is a problem with the defaults.

So, the value of the defaults can be the same as before, not really discussing about this.
But pointing that I'd like to have the same logic in a single place, in that way the logic is clear and clean.

Note, my suggestion is to add a default as last healthconfig, and backend will always have that config available.

But, my concern is first to see the feature working, as next step I'll jump more into the details of the code (not yet reviewing code).

Reply:
About this:

The defaults config are set in the operator.
If backend didn't receive any health config, backend set this default.
If frontend didn't receive from backend any health config, frontend set this default.

Otherwise frontend will use the default.

So....I agree to only replace this in one site. We are calculating the errorRate and status in the frontend so it makes sense to have this here, maybe in a future we can receive this "default parameters" from backend config to the frontend. In this point we can remove the healthConfig from operator and backend.

What do you think?

Reply:
> What do you think?

I was thinking that the defaults can be added in the healthconfig as the "last" entry.

So, in case the user doesn't add any config (via operator, or directly using configmap), Kiali should also append the defaults and make it available in the Kiali UI in the same way.

Why ? In that way, user doesn't need to know what are the defaults hardcoded in the code, he/she can see them from the same place.

I guess this wouldn't change too much the impl, just to add the defaults in the config, I think.

Reply:
> I guess this wouldn't change too much the impl, just to add the defaults in the config, I think.

Not so much, I would need to remove some lines.